crates/wash-cli/src/dev.rs

```diff
@@ -87,6 +85,154 @@ impl Drop for HostSubprocess {
     }
 }
 
+/// WADM components (either WebAssembly components or providers) that can be
+/// used to supply requested functionality in WIT interfaces for components under
+/// development.
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum KnownDep {
+    /// wasmCloud keyvalue provider (normally corresponds to `wasi:keyvalue`)
+    KeyValueProvider,
+    /// wasmCloud HTTP server provider (normally corresponds to `wasi:incoming-handler`)
+    HttpServerProvider,
+    /// wasmCloud HTTP client provider (normally corresponds to `wasi:outgoing-handler`)
+    HttpClientProvider,
+    /// wasmCloud NATS messaging provider (normally corresponds to `wasi:messaging`)
+    NatsMessagingProvider,
+    /// wasmCloud NATS messaging provider (normally corresponds to `wasi:blobstore`)
+    BlobstoreFsProvider,
+    /// Custom provider of some interface (whether a provider or component),
+    /// including the necessary information to identify it as a dependency.
+    Custom {
+        import_interfaces: Vec<String>,
+        export_interfaces: Vec<String>,
+        name: String,
+        image_ref: String,
+    },
+}
+
+impl KnownDep {
+    /// Derive which local component should be used given a WIT interface to be satisified
+    fn from_wit_import_package(p: &Package, iface: &Interface) -> Result<Self> {
+        let name = &format!(
+            "{}:{}/{}",
+            p.name.namespace,
+            p.name.name,
+            iface.name.as_ref().context("interface missing name")?,
+        );
+        match name.split_once('@') {
+            Some(("wasi:keyvalue/atomics", _)) | Some(("wasi:keyvalue/store", _)) => {
+                Ok(KnownDep::KeyValueProvider)
+            }
+            _ => bail!("unsupported WIT interface [{name}]"),
+        }
+    }
```

Comment:
this could also be the blobstore or NATS messaging provider right?

Reply:
Sorry PR was a bit early this part wasn't all filled in yet!